test/test_diffusion.py

```diff
@@ -70,24 +72,35 @@ class HeatProblem:
     sym_u: prim.Expression
 
 
-def get_decaying_cosine(dim, alpha):
+def get_decaying_trig(dim, alpha):
     # 1D: u(x,t) = exp(-alpha*t)*cos(x)
-    # 2D: u(x,y,t) = exp(-2*alpha*t)*cos(x)*cos(y)
-    # 3D: u(x,y,z,t) = exp(-3*alpha*t)*cos(x)*cos(y)*cos(z)
+    # 2D: u(x,y,t) = exp(-2*alpha*t)*sin(x)*cos(y)
+    # 3D: u(x,y,z,t) = exp(-3*alpha*t)*sin(x)*sin(y)*cos(z)
```

Comment:
Would it make sense to also test inhomogeneous Dirichlet/Neumann (not just homogeneous)?

Reply:
Added.